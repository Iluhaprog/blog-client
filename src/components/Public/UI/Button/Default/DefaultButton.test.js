import {fireEvent, render, waitFor} from '@testing-library/react';
import {DefaultButton} from './index';

test('Should render DefaultButton', () => {
  const {asFragment} = render(
    <DefaultButton>
      ok
    </DefaultButton>
  );
  expect(asFragment()).toMatchSnapshot();
});

test('Should render DefaultButton', async () => {
  const click = jest.fn();
  const {container} = render(
    <DefaultButton onClick={click}>
      ok
    </DefaultButton>
  );

  fireEvent.click(container.querySelector('button'));
  await waitFor(() => {
    expect(click).toBeCalledTimes(1);
  });
});
