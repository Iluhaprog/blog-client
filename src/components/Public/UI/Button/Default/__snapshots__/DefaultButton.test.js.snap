// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render DefaultButton 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ hUrxdp"
  >
    ok
  </button>
</DocumentFragment>
`;
